rotkehlchen/premium/premium.py

```diff
@@ -341,41 +353,66 @@ def upload_data(
             our_hash: str,
             last_modify_ts: Timestamp,
             compression_type: Literal['zlib'],
-    ) -> dict:
-        """Uploads data to the server and returns the response dict. We upload the encrypted
-        database as a file in an http form.
+    ) -> None:
+        """Uploads data to the server. We upload the encrypted database in chunks via an http form.
 
         May raise:
         - RemoteError if there are problems reaching the server or if
         there is an error returned by the server
         - PremiumAuthenticationError if the given key is rejected by the Rotkehlchen server
         """
-        data = self.sign(
-            'backup',
-            original_hash=our_hash,
-            last_modify_ts=last_modify_ts,
-            index=0,
-            length=len(data_blob),
-            compression=compression_type,
-        )
+        total_size, upload_id = len(data_blob), None
+        chunk_count = total_size // UPLOAD_CHUNK_SIZE + 1
+        for offset in range(0, total_size, UPLOAD_CHUNK_SIZE):
+            chunk_end = min(offset + UPLOAD_CHUNK_SIZE, total_size)
+            if self.msg_aggregator is not None:
+                self.msg_aggregator.add_message(
+                    message_type=WSMessageType.DATABASE_UPLOAD_PROGRESS,
+                    data={
+                        'type': str(DBUploadStatusStep.UPLOADING),
+                        'current_chunk': offset // UPLOAD_CHUNK_SIZE + 1,
+                        'total_chunks': chunk_count,
+                    },
+                )
```

Comment:
probably you can do this


```suggestion
        for chunk_idx, offset in enumearte(range(0, total_size, UPLOAD_CHUNK_SIZE)):
            chunk_end = min(offset + UPLOAD_CHUNK_SIZE, total_size)
            if self.msg_aggregator is not None:
                self.msg_aggregator.add_message(
                    message_type=WSMessageType.DATABASE_UPLOAD_PROGRESS,
                    data={
                        'type': str(DBUploadStatusStep.UPLOADING),
                        'current_chunk': chunk_idx,
                        'total_chunks': chunk_count,
                    },
                )
```

and avoid the extra division